repositories/decisions_repository.go

```diff
@@ -71,6 +72,42 @@ func (repo *MarbleDbRepository) DecisionWithRuleExecutionsById(ctx context.Conte
 	return decisions[0], nil
 }
 
+func (repo *MarbleDbRepository) DecisionsByOutcomeAndScore(ctx context.Context, exec Executor,
+	scenarioID string, begin, end time.Time,
+) ([]models.DecisionsByVersionByOutcome, error) {
+	decisionQuery := NewQueryBuilder().
+		Select("outcome, scenario_version, score").
+		From(dbmodels.TABLE_DECISIONS).Where(squirrel.GtOrEq{
+		"d.created_at": begin.String(),
+	}).Where(squirrel.LtOrEq{
+		"d.deleted_at": end.String(),
+	})
+	phantomDecisionQuery := NewQueryBuilder().
+		Select("outcome, scenario_version, score").
+		From(dbmodels.TABLE_PHANTOM_DECISIONS).Where(squirrel.GtOrEq{
+		"d.created_at": begin.String(),
+	}).Where(squirrel.LtOrEq{
+		"d.deleted_at": end.String(),
+	})
+	query, err := WithUnionAll(decisionQuery, phantomDecisionQuery)
+	if err != nil {
+		return nil, err
+	}
+	unionQuery, _, _ := query.ToSql()
```

Comment:
By doing this and reusing unionQuery directly below, you're losing the params passed as arguments (begin.String() and end.string(), twice).
Instead, you can:
- do the aggregation (group by) in each subquery and then UNION ALL them
- or kill line 96 and replace `From(unionQuery + " AS q")` below by `FromSelect(query)` (see https://github.com/checkmarble/marble-backend/blob/main/repositories/case_repository.go#L31 for instance)

Reply:
=> gotta run this endpoint/query for real locally before we merge it, because I think as it is it returns an error

Reply:
this is the corresponding raw sql query I tested locally without error
````sql
select q.outcome,q.scenario_version, q.score, Count(q.outcome) as total from (select outcome, scenario_version, score from decisions where created_at>='2011-01-01' and created_at<='2012-01-01' union all select outcome, scenario_version, score from phantom_decisions where created_at>='2011-01-01' and created_at<='2012-01-01') as q group by scenario_version,outcome,score;
````

Reply:
ok, apologies I understand.
Then however that means that the begin/end parameters end up as raw sql in the query, which is not great for another reason. In this case there is no risk of SQL injection of course (which is the n°1 argument for passing params as args rather than code them into the query), but doing so still helps us get proper statistics on queries usage, execution plans, runtime stats etc. (it also helps the DB reuse query plans, but that's perhaps less impactful).

Reply:
Hmm as a matter of fact I'm not able to reproduce your output with `where created_at>='2011-01-01'` 🤔  (using either version)